internal/app/machined/pkg/system/services/udevd.go

```diff
@@ -63,21 +63,27 @@ func (c *Udevd) DependsOn(r runtime.Runtime) []string {
 
 // Runner implements the Service interface.
 func (c *Udevd) Runner(r runtime.Runtime) (runner.Runner, error) {
-	// Set the process arguments.
-	args := &runner.Args{
-		ID: c.ID(r),
-		ProcessArgs: []string{
-			"/sbin/udevd",
-			"--resolve-names=never",
-		},
-	}
-
 	debug := false
 
 	if r.Config() != nil {
 		debug = r.Config().Debug()
 	}
 
+	processArgs := []string{
+		"/sbin/udevd",
+		"--resolve-names=never",
+	}
+
+	if debug {
```

Comment:
don't we start it before the machine config is acquired?

Reply:
yes, but once machine is up, it should have the debug mode, so next boot should provide this.

Reply:
but still we start before `loadConfig`, so unless a service is restarted, I don't see how it would work :shrug:

Reply:
ahh I see the problem, I'll close for now then :sweat_smile: , just wanted to see if it makes easier for others to debug, I usually build a new installer with the `--debug` added :sweat_smile: